### PR TITLE
Bug 1052227 - Move inapp pay test to packaged app to avoid mochitest err...

### PR DESCRIPTION
**Pull request by ferjm**
...ors. r=kumar

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=62888ea354d42d571faa73707166425afa9de9cb

File: dev_apps/in_app_pay_test/js/app.js [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+          };
+          req.onerror = () => {
+            reject('Payment canceled ' + req.error.name);
+          };
+        });
+      }).then((transactionID) => {
+        App.writeLog('Payment done!');
+        App.verifyTransaction(transactionID, 0);
+      }).catch((error) => {
+        App.writeLog('Oh crap ' + error);
```
Comment (kumar303) [inline dev_apps/in_app_pay_test/js/app.js]: heh

File: dev_apps/in_app_pay_test/js/app.js [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+      });
+    },
+
+    verifyTransaction(transactionID, retries) {
```
Comment (kumar303) [inline dev_apps/in_app_pay_test/js/app.js]: this syntax looks weird to me. Did you mean ``verifyTransaction: function() ...`` ?

File: dev_apps/in_app_pay_test/js/app.js [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+      logContent.textContent += '\n' + msg;
+    },
+
+    clearLog: function() {
+      logContent.textContent = '';
+    },
+
+    refresh: function() {
+      getJWT(server.value).then((jwt) => {
+        jwtContent.textContent = JSON.stringify(jwt);
```
Comment (kumar303) [inline dev_apps/in_app_pay_test/js/app.js]: where is ``jwtContent`` defined?

**Comment by kumar303** [comment]
r+ (with comments)

I find it a little awkward to move so much in-app tester code into gaia since maintenance might be tricky. Could this packaged app just be an iframe into the paas site? I suppose another option could be to commit the packaged app change to the [main repo](https://github.com/kumar303/inapp-pay-test) and add it as a submodule to gaia.

File: dev_apps/in_app_pay_test/js/app.js (hunk 3 above)
Comment (kumar303) [inline dev_apps/in_app_pay_test/js/app.js]: ok, I think I see. The code would read better to me if you attached the doc references to a top-level var like ``pageDocs.jwtContent.textContent = ...``

File: dev_apps/in_app_pay_test/js/app.js [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+      });
+    },
+
+    showLog: function() {
+      log.classList.remove('hidden');
+      jwtPanel.classList.add('hidden');
+      toggle.textContent = 'Show JWT';
+    },
+
+    showJWT: function() {
```
Comment (kumar303) [inline dev_apps/in_app_pay_test/js/app.js]: this would be more consistent as ``showJwt``

**Comment by kumar303** [comment]
I think you can remove all the kiwi images. I don't see them referenced anywhere.

File: dev_apps/in_app_pay_test/index.html [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    <title>In-App Payments Tester</title>
+    <meta charset="utf-8"/>
+    <link rel="stylesheet" type="text/css" href="css/index.css"/>
+  </head>
+  <body>
+    <div id="app">
+      <div id="jwt-input">
+        <select name="server" id="server">
+          <option value="0">
+            marketplace.firefox.com (prod)
```
Comment (kumar303) [inline dev_apps/in_app_pay_test/index.html]: on hindsight, allowing an editable JWT to test payments on production seems like a terrible idea. However, I looked at the corresponding key on the server and it does not seem to be active. Choosing this option will cause an error.

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1e2cb204a191f43739e3a274848e8b30044f357b

**Pull request closed without merging**